src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/Instruction.cs

```diff
@@ -55,6 +55,15 @@ public virtual object GetDebugCookie(LightCompiler compiler)
         {
             return null;
         }
+
+        // throws NRE when o is null
+        protected static void NullCheck(object o)
+        {
+            if (o == null)
+            {
+                o.GetType();
```

Comment:
unused   "o.getClass" is an idiomatic null check in java (since unused getClass is otherwise not considered sideeffecting). 
Unfortunately our JITs seem to emit the actual call, so I need the condition above.

I wonder if this is worth entering a bug...